### DISCUSSION TITLE
Question: use of QueueEvents callbacks across multiple workers

### DISCUSSION THREAD

**Discussion by liquid1982** [taskforcesh/bullmq#1623] [Q&A] [answered]
We'd like to capture and process the `stalled` event from `QueueEvents`, but it is unclear whether it would make more sense to have 1 or 2 dedicated instances to register/execute the callbacks, or if there are no concerns about that on a performance/overhead standpoint for the workers.

For example, for the `QueueScheduler` documentation, there's indication about it:

> It is ok to have as many QueueScheduler instances as you want, just keep in mind that every instance will perform some bookkeeping so it may create some noticeable CPU and IO usage in your Redis instances.

But again, I'd like to hear from the team if there are recommendations around this use case.

Thanks in advance to whoever wants to chime in here!

**Comment by manast** [comment, marked as answer]
The QueueScheduler is not needed anymore, so that answers that question 😅. I do not have any benchmarks on the overhead for QueueEvents, but it is basically listening to Redis streams, which are pretty fast, but if you have a lot of these instances it will probably result in a noticeable overhead. The best is to make some benchmarks for your specific use case and see what results you get.

**Reply by liquid1982** [reply]
Thanks for sharing your thoughts, @manast!
